session/session.go

```diff
@@ -378,44 +371,76 @@ func (s *Session) cidIsWanted(c cid.Cid) bool {
 	return ok
 }
 
-func (s *Session) receiveBlock(ctx context.Context, blk blocks.Block) {
-	c := blk.Cid()
-	if s.cidIsWanted(c) {
-		s.srs.RecordUniqueBlock()
```

Comment:
Note: `s.srs.RecordUniqueBlock()` has now been moved into `updateReceiveCounters()`. Everything else should be the same